compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -2117,4 +2117,22 @@ object messages {
     override def msg: String = hl"""normal case class cannot extend an enum. case $cls in ${cls.owner} is extending enum ${parent.name}."""
     override def explanation: String = ""
   }
+
+  case class NotAnExtractor(tree: tpd.Tree)(implicit ctx: Context) extends Message(NotAnExtractorID) {
+    override def msg: String = hl"""${tree.show} cannot be used as an extractor in a pattern because it is not implemented"""
+    override def kind: String = "Syntax"
+    override def explanation: String =
+      hl"""| ${tree.show} is not a valid extractor for use in pattern matching,
+           | this happens when the method unapply or unapplySeq it is not declared
+           | on it's object companion.
+           | An unapply method take an object and tries to give back the arguments
```

Comment:
This is one use case among others. E.g.
```scala
object PositiveInt {
  def unapply(x: Int) = x >= 0
}

def foo(x: Int) = x match { case PositiveInt() => ...
```
I would replace your explanation and example by something that explain what is a valid extractor object. A brief summary of https://docs.scala-lang.org/tour/extractor-objects.html would be awesome

Reply:
Sorry, I didn't mean to add more use cases but explain how one can write a valid extractor object. E.g.
```scala
def explanation =
  hl"""|An `unapply` method should be defined as follow:
       |  - If it is just a test, return a `Boolean`. For example `case even()`
       |  - If it returns a single sub-value of type T, return an `Option[T]`
       |  - If it returns several sub-values T1,...,Tn, group them in an optional tuple `Option[(T1,...,Tn)]`
       |
       |Sometimes, the number of sub-values isn’t fixed and we would like to return a sequence.
       |For this reason, you can also define patterns through `unapplySeq` which returns `Option[Seq[T]]`.
       |This mechanism is used for instance in pattern `case List(x1, ..., xn)`"""
```

Reply:
Got it! Should we also point that the unapply method should be defined in an object ? E.g.
```
An `unapply` method should be defined in an object as follow:
```
I think of cases where the user uses a plain class instead of an object, then tries to define the method inside the class and still get the error

Reply:
Yes, sounds good!